### PR TITLE
Issue/1823 allow stacktraces without phonenumber (Connect #1823)

### PR DESCRIPTION
**Pull request by stellanl**
#### Before the PR (what is the issue or what needed to be done)
See #1823
#### The solution
Removed the check for phoneNumber parameter
Added storing of androidId parameter
Gave a simpleDateFormat constructor a locale parameter
## Checklist
* [x] Connect the issue
* [x] Test plan
* [x] Copyright header
* [x] Code formatting
* [x] Documentation

### CODE REVIEW

File: GAE/src/com/gallatinsystems/diagnostics/app/web/dto/RemoteExceptionRequest.java [hunk 1]
```diff
@@ -95,13 +98,16 @@ public void setDate(Date date) {
         this.date = date;
     }
 
-    @Override
+    public String getAndroidId() {
+		return androidId;
+	}
```
Comment (muloem) [inline GAE/src/com/gallatinsystems/diagnostics/app/web/dto/RemoteExceptionRequest.java]: The source formatting here and below is a bit off.

File: GAE/src/com/gallatinsystems/diagnostics/domain/RemoteStacktrace.java [hunk 2]
```diff
@@ -87,4 +88,12 @@ public void setErrorDate(Date errorDate) {
         this.errorDate = errorDate;
     }
 
+    public String getAndroidId() {
+		return androidId;
+	}
```
Comment (muloem) [inline GAE/src/com/gallatinsystems/diagnostics/domain/RemoteStacktrace.java]: Also source code formatting here and below is a bit off

Comment (stellanl) [inline GAE/src/com/gallatinsystems/diagnostics/domain/RemoteStacktrace.java]: Oops, had not setup that Eclipse instance to only use spaces...

**Comment by muloem** [comment]
:+1: code reviewed!

**Pull request merged**